sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -256,6 +315,162 @@ func resourceSumologicMonitorsLibraryMonitor() *schema.Resource {
+		"time_range": &timeRangeSchema,
+		"alert": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+		"resolution": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+	}),
+	"warning": nested(true, schemaMap{
+		"time_range": &timeRangeSchema,
+		"alert": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+		"resolution": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+	}),
+}
+
+var metricsStaticTriggerConditionSchema = map[string]*schema.Schema{
+	"critical": nested(true, schemaMap{
+		"time_range":      &timeRangeSchema,
+		"occurrence_type": &occurrenceTypeSchema,
+		"alert": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+		"resolution": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+	}),
+	"warning": nested(true, schemaMap{
+		"time_range":      &timeRangeSchema,
+		"occurrence_type": &occurrenceTypeSchema,
+		"alert": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+		"resolution": nested(false, schemaMap{
+			"threshold":      &thresholdSchema,
+			"threshold_type": &thresholdTypeSchema,
+		}),
+	}),
+}
+
+var logsOutlierTriggerConditionSchema = map[string]*schema.Schema{
+	"field": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"direction": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		ValidateFunc: validation.StringInSlice([]string{"Both", "Up", "Down"}, false),
+	},
+	"critical": nested(true, schemaMap{
+		"window":      &windowSchema,
+		"consecutive": &consecutiveSchema,
+		"threshold":   &thresholdSchema,
+	}),
+	"warning": nested(true, schemaMap{
+		"window":      &windowSchema,
+		"consecutive": &consecutiveSchema,
+		"threshold":   &thresholdSchema,
+	}),
+}
+
+var metricsOutlierTriggerConditionSchema = map[string]*schema.Schema{
+	"direction": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		ValidateFunc: validation.StringInSlice([]string{"Both", "Up", "Down"}, false),
+	},
+	"critical": nested(true, schemaMap{
+		"baseline_window": &baselineWindowSchema,
+		"threshold":       &thresholdSchema,
+	}),
+	"warning": nested(true, schemaMap{
+		"baseline_window": &baselineWindowSchema,
+		"threshold":       &thresholdSchema,
+	}),
+}
+
+var logsMissingDataTriggerConditionSchema = map[string]*schema.Schema{
+	"time_range": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validation.StringInSlice([]string{"5m", "-5m", "10m", "-10m", "15m", "-15m", "30m", "-30m", "60m", "-60m", "1h", "-1h", "3h", "-3h", "6h", "-6h", "12h", "-12h", "24h", "-24h", "1d", "-1d"}, false),
```

Comment:
nit: I was thinking there should be a way to define a custom `ValidationFunc` (as a subtype of `SchemaValidationFunc` I guess? https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema#SchemaValidateFunc ) separately and then re-use it across the various trigger conditions for fields where the validation will be common/shared (like timerange) - but I couldn't find an obvious way with a quick search 🤔  either way shouldn't really matter too much

Reply:
Ah, thanks for pointing out. Looks like I forgot to replace the block with  
```
"time_range":      &timeRangeSchema,
```
here (and in the next schema), like I did for the rest. Will fix.